sympy/integrals/prde.py

```diff
@@ -917,16 +918,96 @@ def parametric_log_deriv_heu(fa, fd, wa, wd, DE, c1=None):
     return (Q*N, Q*M, v)
 
 
+def linear_relations_in_Q(F):
+    """
+    Parameters
+    ==========
+
+    F: A list of expressions (not necessarily Poly)
+
+    Returns
+    =======
+
+    The rational linear relations
+
+    Examples
+    ========
+
+    """
+    # find a field containing all the fractions
+    # 'field=True' covers the non-algebraic case
+    FF, gens = sfield(F, field=True, extension=True)
+    K = FF.domain
+
+    denom = reduce(lambda i, j: i.lcm(j), [gen.denom for gen in gens])
```

Comment:
`reduce` is no longer a built-in function in Python 3. It has to be imported from functools. (Its use is probably being discouraged.)